### ISSUE TITLE
hobo Waschmaschine nachziehen

### ISSUE THREAD

**Issue by sidey79** [RFD-FHEM/RFFHEM#840]
Die Korrekuren von Hobo übernehmen.

https://gl.petatech.eu/root/HomeBot/-/commit/52e28f2b545d2c6a5c126bcd06ddd25a4f496863
und
https://gl.petatech.eu/root/HomeBot/-/commit/af8e450f08289b4b202b05a9f1da50b3a1813309

**Comment by HomeAutoUser** [comment]
Was ist der Grund für https://gl.petatech.eu/root/HomeBot/-/commit/52e28f2b545d2c6a5c126bcd06ddd25a4f496863 ?Ich vermisse da eine Beschreibung um was es geht.

`initial wash-diff for 00-modules`
sagt mir wenig aus.

https://gl.petatech.eu/root/HomeBot/-/commit/af8e450f08289b4b202b05a9f1da50b3a1813309

hier geht es vermutlich darum,
- das return undef behoben wird
- das foreach durch for geändert wird.

WAS ist der Grund, das solche Maßnahmen gerade vollzogen werden?
Jeder der mit einem Modul beginnt, schaut in die Anleitung von PERL und findet soetwas als Beispiele dort wieder.

**Comment by sidey79** [comment]
Da werden die gröbsten Fehler entfernt, die mit perlcritic aufgefallen sind.

Und ich glaube das Einrücken ist auch korrigiert , daher würde auch jede Zeile verändert.

Naja, sind best practies es so zu tun

**Comment by sidey79** [comment]
Tabs sind vermutlich auch so ein Thema

Mit folgendem Befehl lassen sich die Tabs durch leerzeichen ersetzen:
 expand -i -t 4 ./FHEM/lib/SD_Protocols.pm | sponge ./FHEM/lib/SD_Protocols.pm

**Comment by elektron-bbs** [comment]
Das ersetzt aber wahrscheinlich alle Tabs durch Leerzeichen, oder?
Perlcritic bemängelt ja nur Tabs, die nicht am Zeilenanfang stehen.

**Comment by sidey79** [comment]
Ja das stimmt,
Gefühl passt das mit den Tabs oft aber nicht in den verschiedenen Editoren.

**Comment by HomeAutoUser** [comment]
Verstehe ich das richtig, das hier
`expand -i -t 4 ./FHEM/lib/SD_Protocols.pm | sponge ./FHEM/lib/SD_Protocols.pm`
das ein Ansatz war, um in der Datei SD_Protocols.pm die Tabs durch Leerzeichen zu ersetzen wegen Perlcritic ?

**Comment by sidey79** [comment]
Ja, das war ein Ansatz dass jeder Tab durch vier Leerzeichen ersetzt wird.

**Comment by HomeAutoUser** [comment]
Wieso unbedingt 4 Leerzeichen? Das ist je Benutzer konfigurierbar. Manche Benutzer haben 2. Ein anderer vielleicht 6? Mit welcher Einstellung arbeitest du?

**Comment by sidey79** [comment]
4 ist glaube ich das, worauf wir uns schon einmal geeinigt hatten.

**Comment by elektron-bbs** [comment]
Nö, ich arbeite aktuell mit 2 Leerzeichen pro Tab.

**Comment by HomeAutoUser** [comment]
*smile*
Klassischer Fall, das es eigentlich unnötig ist das als Argument PerCritic zu verarbeiten.

Ich hätte mir die Arbeit gemacht aber da muss Einigkeit herrschen. Ich bilde mir ein, es gibt einmal schon ein issues wo wir 2 Leerzeichen bezifferten.

**Comment by sidey79** [comment]
2 4 oder 8 auf irgendwas hatten wir uns schon einmal geeinigt.
Gefunden habe ich es nicht.

Da wir aktuell schon nicht mit dem gleichen Wert arbeitet, erklärt vermutlich, wieso es doch immer Mal wieder Überraschungen gibt

**Comment by HomeAutoUser** [comment]
Ich habe mir soeben die Mühe gemacht und nochmal geschaut. Das einzige was wir mal geschrieben hatten und auch verankert hatten ist wie folgt:

https://github.com/RFD-FHEM/RFFHEM/pull/588#issuecomment-491985581

> Da wir aktuell schon nicht mit dem gleichen Wert arbeitet, erklärt vermutlich, wieso es doch immer Mal wieder Überraschungen gibt

ABER

das Problem ist
1) das die Stellen der Bearbeitung an verschiedenen System passiert | Bsp: - Rechner Daheim, Rechner Vater unsw.
2) das die Systeme | OS | Editoren , teilweise auch unterschiedliche Einstellungen verankert haben

Ich denke, man eine Anzahl festlegen aber diese kann denn noch zu Unterschieden führen weil ich nicht jedes mal die Tabulatorenbreite bei jedem PC oder Geräte prüfe.

Vorschlag:
Die Datei wird nach einem festen Stand überarbeitet, das die Tabs verschwieden und somit das PerlCritic erledigt ist aber danach kann es auch ab und zu zu Differenzen kommen.

**Comment by elektron-bbs** [comment]
Perlcritic mäkelt doch nicht generell an den Tabs herum. Führende Tabs sind OK, nur Tabs innerhalb der Codezeilen werden bemängelt. Ob ich dann dort 2 oder 4 Leerzeichen setze, ist doch prinzipiell egal. Das beste Beispiel ist doch die SD_ProtocolData.pm:

```
				name            => 'TR60C-1',
				comment         => 'Remote control for example Westinghouse Bendan 77841B',
				id              => '104',
				knownFreqs      => '433.92',
				one             => [-1,2],  #  -380,760
				zero            => [-2,1],  #  -760,380
				start           => [-25,1], # -9500,380
				clockabs        => 380,
				format          => 'twostate',
```

Vorn sind Tabs und zwischen Keys und Values sind Leerzeichen.

**Comment by sidey79** [comment]
Wenn es bei dem einen zwei sind und bei dem anderen 8 dann hat das durchaus eine Optische Auswirkung wenn ich vier Tabs hintereinander mache
